doc/source/ray-tracing.rst

```diff
@@ -0,0 +1,107 @@
+
+.. code-block:: shell
+
+    pip install opentelemetry-api==1.0.0rc1
+    pip install opentelemetry-sdk==1.0.0rc1
+
+To enable tracing, you must provide a tracing startup hook with a function that will set up the Tracer Provider, Remote Span Processors, and Additional Instruments. The tracing startup hook is expected to be a function that will be called with no args or kwargs. This hook needs to be available in the Python environment of all the worker processes.
+
+Tracer Provider
+~~~~~~~~~~~~~~~
+This configures how to collect traces. View the TracerProvider API `here <https://open-telemetry.github.io/opentelemetry-python/sdk/trace.html#opentelemetry.sdk.trace.TracerProvider>`__.
+
+Remote Span Processors
+~~~~~~~~~~~~~~~~~~~~~~
+This configures where to export traces to. View the SpanProcessor API `here <https://open-telemetry.github.io/opentelemetry-python/sdk/trace.html#opentelemetry.sdk.trace.SpanProcessor>`__.
+
+Users who want to experiment with tracing can configure their remote span processors to export spans to a local JSON file. Serious users developing locally can push their traces to Jaeger containers via the `Jaeger exporter <https://open-telemetry.github.io/opentelemetry-python/exporter/jaeger/jaeger.html>`_.
+
+
+Additional Instruments
+~~~~~~~~~~~~~~~~~~~~~~
+If you are using a library that has built-in tracing support, the ``setup_tracing`` function you provide should also patch those libraries. You can find more documentation for the instrumentation of these libraries `here <https://github.com/open-telemetry/opentelemetry-python-contrib/tree/main/instrumentation>`_.
+
+Below is an example tracing startup hook that sets up the default tracing provider, exports spans to files in ``/tmp/spans``, and does not have any additional instruments.
+ 
+.. code-block:: python
+
+  import ray
+  import os
+  from opentelemetry import trace
+  from opentelemetry.sdk.trace import TracerProvider
+  from opentelemetry.sdk.trace.export import (
+      ConsoleSpanExporter,
+      SimpleExportSpanProcessor,
+  )
+  
+  
+  def setup_tracing() -> None:
+      # Sets the tracer_provider. This is only allowed once per execution
+      # context and will log a warning if attempted multiple times.
+      trace.set_tracer_provider(TracerProvider())
+      trace.get_tracer_provider().add_span_processor(
+          SimpleExportSpanProcessor(
+              ConsoleSpanExporter(
+                  out=open(f"/tmp/spans/{os.getpid()}.json", "a")
+                  )
+          )
+      )
+
+
+For open-source users who want to experiment with tracing, Ray has a default tracing startup hook that exports spans to the folder ``/tmp/spans``. To run using this default hook, you can run the following code sample to set up tracing and trace a simple Ray task.
+
+.. tabs::
+    .. group-tab:: ray start
+
+        .. code-block:: shell
+
+            $ ray start --head --tracing-startup-hook=ray.util.tracing.enable_tracing:setup_tracing
```

Comment:
make sure to change the utils enable_tracing to tests.enable tracing to convey that this is really just used for proof of concept not for production.

Reply:
done